lumicks/pylake/detail/tests/test_plotting.py

```diff
@@ -0,0 +1,155 @@
+
+
+def test_get_axes():
+    fig, (ax1, ax2) = plt.subplots(2)
+    ih1 = ax1.imshow(np.empty(shape=(2, 2)))
+    ih2 = ax2.imshow(np.empty(shape=(2, 2)))
+
+    ax = get_axes(axes=ax1)
+    assert ax is ax1
+    ax = get_axes()
+    assert ax is ax2
+    ax = get_axes(image_handle=ih1)
+    assert ax is ax1
+    ax = get_axes(image_handle=ih2)
+    assert ax is ax2
+    ax = get_axes(axes=ax1, image_handle=ih1)
+    assert ax is ax1
+    with pytest.raises(
+        ValueError, match="Supplied image_handle with a different axes than the provided axes"
+    ):
+        ax = get_axes(axes=ax1, image_handle=ih2)
+
+    plt.close(fig)
+
+
+def test_show_image():
+    im1 = np.empty(shape=(2, 2))
+    im2 = np.empty(shape=(3, 3))
+
+    fig, (ax1, ax2) = plt.subplots(2)
+    ih1 = ax1.imshow(im1)
+
+    with pytest.raises(
+        ValueError, match="Supplied image_handle with a different axes than the provided axes"
+    ):
+        show_image(im1, image_handle=ih1, axes=ax2)
+
+    # Test if image handle is used for plotting
+    ih = show_image(im2, image_handle=ih1)
+    assert ih is ih1
+    np.testing.assert_equal(ih.get_array(), im2)
+
+    # Test if image handle is used for plotting and axes is ignored (no kwargs forwarded to imshow)
+    ih = show_image(im1, image_handle=ih1, axes=ax1, url="NOT_FORWARED_KWARG")
+    assert ih is ih1
+    np.testing.assert_equal(ih.get_array(), im1)
+    assert ih.get_url() is None
+
+    # Test if axes is used for plotting (i.e. new image handle is created)
+    ih = show_image(im2, axes=ax1)
+    assert ih.axes is ax1
+    assert ih is not ih1
+    np.testing.assert_equal(ih.get_array(), im2)
+
+    # Test if current axes is used for plotting
+    ih = show_image(im1)
+    assert ih.axes is ax2
+    np.testing.assert_equal(ih.get_array(), im1)
+
+    # Test if kwargs are forwarded to plt.imshow()
+    ih = show_image(im1, axes=ax1, url="FORWARDED_KWARG")
+    assert ih.get_url() == "FORWARDED_KWARG"
+
+    # Test if ColorAdjustment is applied to image_handle
+    ih = show_image(im2, axes=ax1, adjustment=ColorAdjustment(1.5, 2, mode="absolute"))
+    assert ih.norm.vmin == 1.5
+    assert ih.norm.vmax == 2
+
+    plt.close(fig)
+
+
+def _plot_interface_parameters(image=False, stack=False):
+    """Get a list of parameters the signature of a plot function is supposed to have
+
+    Parameters
+    ----------
+    image : bool
+        The plot function is supposed to support plotting of images
+    stack : bool
+        The plot function is supposed to support selecting frames from image stacks
+
+    Returns
+    -------
+    List[Tuple]
+        A list of (name, kind) of expected parameters
+    """
+    supported_interfaces = ["general"]
+    if image:
+        supported_interfaces.append("image")
+    if stack:
+        supported_interfaces.append("stack")
+
+    POSITIONAL_OR_KEYWORD = Parameter.POSITIONAL_OR_KEYWORD
+    KEYWORD_ONLY = Parameter.KEYWORD_ONLY
+    VAR_KEYWORD = Parameter.VAR_KEYWORD
```

Comment:
I would get rid of this part and just use `Parameter.VALUE` in the list below. It's a bit of repetition but I think it's fine

Reply:
removed.